### PR TITLE
#182 Show the start page with title and message defined for template

### LINKED ISSUE #182: Add start message and submission message for Templates
**Updated description on 29/12 after creation of Epic#24**

## Scope
- Add two new TEXT fields to `template` table: `start_message`, `submission_message`
   - Submission message should have a default: "Thank you... etc.", but NOT the start message
- Write start and submission messages for existing templates (put in `insertData.js`)

### PR DESCRIPTION
**Pull request by nmadruga**
Fix #182 

Using [Backend-#182F](https://github.com/openmsupply/application-manager-server/pull/162).

## Changes
- Layout on start application page
- Fetches `startTitle` and `startMessage` for template to be displayed on start application page.
- Add every string as constant to the constants file.
- Show application page if start message is defined.
- [Existing behaviour] Create the application when 'Get started' button is pressed.

## To do next:
- [#232] If `startTitle` or `startMessage` not defined - go straight to start page OR last invalid (or incomplete) page.
- [#236] Show progress on start page of Draft application:
  - Show progress on each section and validation icons.
  - Hide 'Get started' button when application already exists.
  - Show 'Resume' button on the last section with invalid (or incomplete) questions.

### CODE REVIEW

File: src/components/Application/ApplicationStart.tsx [hunk 1]
```diff
@@ -10,27 +11,62 @@ export interface ApplicationStartProps {
...
-              ))}
-          </List>
-          <Button content={type.name} onClick={handleClick} />
-        </Segment>
-      )}
-      {!type && <Label content="No Application" />}
-    </Container>
+  return template ? (
+    <Segment.Group style={{ backgroundColor: 'Gainsboro', display: 'flex' }}>
+      <Button as="href" icon="angle left" label={{ content: name, color: 'grey' }} />
```
Comment (CarlosNZ) [inline src/components/Application/ApplicationStart.tsx:19]: Not really a major thing right now, but I feel that a "label" element might not be the best way to represent this type of link. Also, note that the Designs show this name in plural (i.e. "User Registrations"), so maybe we need a better way to represent this. (Perhaps for now we could have `name + " applications"`, so it'll show as "User Registration applications")

File: src/components/Application/ApplicationStart.tsx [hunk 2]
```diff
@@ -10,27 +11,62 @@ export interface ApplicationStartProps {
...
+                <Message.Header>
+                  <Icon name="info circle" />
+                  {startTitle}
+                </Message.Header>
+                {startMessage && (
+                  <Message.List>
+                    {startMessage.split(/\r?\n/).map((str, index) => (
+                      <Message.Item key={`start-item-${index}`}>{str}</Message.Item>
+                    ))}
+                  </Message.List>
```
Comment (CarlosNZ) [inline src/components/Application/ApplicationStart.tsx:59]: I don't think we should convert new lines into lists like this. If the template wants to define a list it should use a markdown list. For now, I'd suggest keeping it as simple as possible and just displaying the message as-is, then we can improve the layout when we add markdown interpretation. (See also my back-end comments regarding title and message seperation -- i.e. I don't think we need it)

File: src/utils/constants.ts [hunk 3]
```diff
@@ -25,10 +26,14 @@ export default {
...
+  TITLE_APPLICATION_FORM: 'application form',
   TITLE_APPLICATION_SUBMIT: 'Review & Submit',
   TITLE_COMPANY_PLACEHOLDER: 'ABC Company',
+  TITLE_INTRODUCTION: 'Introduction',
   TITLE_REVIEW_SUMMARY: 'Review Summary',
-  TITLE_PROGRESS: 'Introduction',
+  TITLE_STEPS: 'Steps to complete',
   STAGE_PLACEHOLDER: 'Assessment',
+  SUBTITLE_APPLICATION_STEPS:
+    'The following sterps will need to be completed before the form can be submitted',
```
Comment (CarlosNZ) [inline src/utils/constants.ts:37]:
```suggestion
    'The following steps will need to be completed before the form can be submitted',
```

File: src/utils/hooks/useLoadTemplate.tsx [hunk 4]
```diff
@@ -46,14 +46,23 @@ const useLoadTemplate = (props: useLoadTemplateProps) => {
       return
     }
 
-    const { id, code, name, templateSections } = template
+    const {
+      id,
+      code,
+      name,
+      startTitle,
+      startMessage,
```
Comment (CarlosNZ) [inline src/utils/hooks/useLoadTemplate.tsx:54]: See my back-end comments -- I don't think we need seperate Title and Message fields. Markdown should be able to handle it all as one "Message" field.

**Review by CarlosNZ — CHANGES_REQUESTED** [review]

File: src/components/Application/ApplicationStart.tsx (hunk 1 above)
Comment (nmadruga) [reply src/components/Application/ApplicationStart.tsx:19]: Sure, I agree. I'll make this improvement now.

Comment (CarlosNZ) [inline src/components/Application/ApplicationStart.tsx:19]: Just noticed this also: the `as="href"` is causing this error:
Screen Shot 2021-01-06 at 10 26 15 AM

I'm not sure what this prop is doing -- seems like you can just remove it?

Comment (nmadruga) [reply src/components/Application/ApplicationStart.tsx]: Done: [894c2fe](https://github.com/openmsupply/application-manager-web-app/pull/233/commits/894c2fe49fbfe370a99c2a306203b985fec502e1)

File: src/components/Application/ApplicationStart.tsx (hunk 2 above)
Comment (nmadruga) [reply src/components/Application/ApplicationStart.tsx:63]: No problem. Now will be using markdown instead. Updating...

Comment (nmadruga) [reply src/components/Application/ApplicationStart.tsx]: Done: [11371fd](https://github.com/openmsupply/application-manager-web-app/pull/233/commits/11371fd5acbe75192639c4141832e7a421eb6c86)

File: src/components/Application/ApplicationStart.tsx (hunk 1 above)
Comment (nmadruga) [reply src/components/Application/ApplicationStart.tsx]: It's fixed in the next PR: https://github.com/openmsupply/application-manager-web-app/pull/234/files?file-filters%5B%5D=.png&file-filters%5B%5D=.ts&file-filters%5B%5D=.tsx#diff-4d0ea65adc1a078ca90ef8e261c5544189bd5d38de5c6e051f09f963a6cce354

**Review by CarlosNZ — APPROVED** [review]

**Pull request merged**